src/aiod/authorisation/authorisation.py

```diff
@@ -0,0 +1,80 @@
+import os
+from keycloak import KeycloakOpenID
+
+from aiod.config.settings import server_url, client_id, realm
+
+keycloak_openid = KeycloakOpenID(
+    server_url=server_url,
+    client_id=client_id,
+    realm_name=realm,
+)
+
+
+def login(username: str, password: str) -> None:
+    """
+    Logs in the user with the provided username and password.
+
+    Args:
+        username (str): The username of the user.
+        password (str): The password of the user.
+
+    Raises:
+        MissingCredentialsError: If the username or password is missing.
+        MissingTokenError: If Keycloak returns an invalid token.
+    """
+    if username is None or password is None:
+        raise MissingCredentialsError(
+            "Username and/or password missing! Please provide your credentials and try again."
+        )
+    token = keycloak_openid.token(username, password)
+
+    if not token:
+        raise MissingTokenError("Keycloak returned an invalid token!")
+    os.environ["ACCESS_TOKEN"] = token["access_token"]
+    os.environ["REFRESH_TOKEN"] = token["refresh_token"]
+
+
+def logout() -> None:
+    """
+    Logs out the current user.
+
+    Raises:
+        MissingTokenError: If the stored refresh token is empty.
+    """
+    refresh_token = get_refresh_token()
+    if not refresh_token:
+        raise MissingTokenError(
+            "The stored refresh token is empty! Please try to login again."
+        )
```

Comment:
Throwing an error here does not make a lot of sense to me. The user wants to log out, and it turns out they are not even logged in. Task failed successfully? At most I would make it optional.

Reply:
Indeed. It makes sense.